src/training/train.py

```diff
@@ -69,20 +79,60 @@ def train_one_epoch(model, data, epoch, optimizer, scaler, scheduler, args, tb_w
     end = time.time()
     for i, batch in enumerate(dataloader):
         step = num_batches_per_epoch * epoch + i
-        
+
         if not args.skip_scheduler:
             scheduler(step)
 
-        images, texts = batch
+        images, texts, text_input_masks = batch['image'], batch['text'], batch['text_input_mask']
         images = images.to(device=device, dtype=cast_dtype, non_blocking=True)
         texts = texts.to(device=device, non_blocking=True)
+        text_input_masks = text_input_masks.to(device=device, non_blocking=True)
+
+        if 'itm' in batch:
+            itm_images = batch['itm']['image']
+            itm_labels = batch['itm']['label']
+            itm_images = itm_images.to(device=device, dtype=cast_dtype, non_blocking=True)
+            itm_labels = itm_labels.to(device=device, non_blocking=True)
+
+        if 'mlm' in batch:
+            text_masked = batch['mlm']['input_ids']
+            text_masked_labels = batch['mlm']['labels']
+            text_masked = text_masked.to(device=device, non_blocking=True)
+            text_masked_labels = text_masked_labels.to(device=device, non_blocking=True)
 
         data_time_m.update(time.time() - end)
         optimizer.zero_grad()
 
         with autocast():
-            image_features, text_features, logit_scale = model(images, texts)
-            total_loss = loss(image_features, text_features, logit_scale)
+            if is_flava:
+                assert 'mlm' in batch, 'FLAVA pretraining requires MLM inputs.'
+                assert 'itm' in batch, 'FLAVA pretraining requires ITM inputs.'
+
+                image_features, \
+                text_features, \
+                logit_scale, \
+                text_masked_recon, \
+                itm_logits, \
+                mm_text_masked_recon = model(
```

Comment:
Then the batch here goes directly to the model as it is. The loss is returned and then passed down the line. This will make sure the trainer is simplified.